packages/bot-web-ui/src/stores/download-store.js

```diff
@@ -0,0 +1,118 @@
+import { action } from 'mobx';
+import { localize } from '@deriv/translations';
+import { log_types } from '@deriv/bot-skeleton/src/constants/messages';
+
+export default class DownloadStore {
+    constructor(root_store) {
+        this.root_store = root_store;
+    }
+
+    getSuccessJournalMessage = (message, extra) => {
+        const { profit, sold_for, longcode, transaction_id } = extra;
+        switch (message) {
+            case log_types.LOAD_BLOCK: {
+                return localize('Blocks are loaded successfully');
+            }
+            case log_types.NOT_OFFERED: {
+                return localize('Resale of this contract is not offered.');
+            }
+            case log_types.PURCHASE: {
+                return localize('Bought: {{longcode}} (ID: {{transaction_id}})', { longcode, transaction_id });
+            }
+            case log_types.SELL: {
+                return localize('Sold for: {{sold_for}}', { sold_for });
+            }
+            case log_types.PROFIT: {
+                return localize('Profit amount: {{profit}}', { profit });
+            }
+            case log_types.LOST: {
+                return localize('Loss amount: {{profit}}', { profit });
+            }
+            default:
+                return null;
+        }
+    };
+
+    @action.bound
+    onClickDownloadTransaction = () => {
+        // Transaction Array
+        const transaction_csv_titles = [
+            [
+                localize('Market'),
+                localize('Reference ID (buy)'),
+                localize('Reference ID (sell)'),
+                localize('Barrier'),
+                localize('Start Time'),
+                localize('Entry Spot'),
+                localize('Entry Spot Time'),
+                localize('Exit Spot'),
+                localize('Exit Spot Time'),
+                localize('Buy Price'),
+                localize('Profit/Loss'),
+            ],
+        ];
+        const transaction_messages = this.root_store.transactions.elements;
+        transaction_messages.map(item => {
+            if (item.type === 'contract') {
```

Comment:
We could use the constants defined in `deriv-app/packages/bot-web-ui/src/constants/transactions.js` here.